### DISCUSSION TITLE
Changing angle of attack from python

### DISCUSSION THREAD

**Discussion by kursatyurt** [su2code/SU2#2028] [Q&A]
I am trying to get polar using my own script.

I don't want to produce a config file for each case as in the polar script.

I have only 2 markers as in the NACA0012 case with a compressible solver. Normally even if there are some restrictions about AoA I expect that the flow angle should change but also did not observe any change in flow angle. The case is steady.

My script is very simple:

```python
#!/usr/bin/env python3
import sys

import numpy as np
import pysu2
import SU2
from mpi4py import MPI


def setup_mpi():
    if "mpi4py.MPI" in sys.modules:
        have_mpi = True
        comm = MPI.COMM_WORLD
        status = MPI.Status()
        myid = comm.Get_rank()
    else:
        have_mpi = False
        comm = None
        status = None
        myid = 0
    return have_mpi, comm, status, myid


class SU2Interface:
    def __init__(self, config_filename):
        self.have_mpi, self.comm, self.status, self.myid = setup_mpi()
        self.FluidSolver = pysu2.CSinglezoneDriver(config_filename, 1, self.comm)
        
    def setAoA(self,angle,marker):
        if marker in self.FluidSolver.GetMarkerIndices().keys():   
            markerID = self.FluidSolver.GetMarkerIndices()[marker]
            self.FluidSolver.SetInletAngle(0,angle)
        self.FluidSolver.BoundaryConditionsUpdate()
    
    def runSolver(self):
        self.comm.barrier()
        # run solver
        self.FluidSolver.Preprocess(0)
        self.FluidSolver.Run()
        self.FluidSolver.Postprocess()
        if self.myid == 0:
            print(f" AOA: {self.FluidSolver.GetOutputValue('AOA')}")
            print(f" LIFT : {self.FluidSolver.GetOutputValue('LIFT')}")
            print(f" DRAG : {self.FluidSolver.GetOutputValue('DRAG')}")
        self.comm.barrier()
        
    def finalize(self):
        self.FluidSolver.Finalize()
        


for angle in np.linspace(0,10,5):
    cfd_interface = SU2Interface('NACA0012.cfg')
    cfd_interface.setAoA(angle,'farfield')
    cfd_interface.run_solver()
    cfd_interface.finalize()
```

**Comment by pcarruscag** [comment]
The inlet angle is for inlet boundaries, not for farfield.
The wrapper doesn't have a function to change farfield direction yet

**Reply by kursatyurt** [reply]
@pcarruscag thank you for your quick response.

Is it possible to change far-field, or restricted? If you guide me on where should I look (options or the driver itself or somewhere else?) I would like to implement it.

**Reply by pcarruscag** [reply]
See the end of CEulerSolver::SetFarfield_AoA

**Comment by ArneVoss** [comment]
I use the translating mesh for this / a similar purpose. Set the farfield to zero and use u, v and w instead. The python interface to update the velocities has just been merged into the development branch (https://github.com/su2code/SU2/pull/2024).

**Reply by kursatyurt** [reply]
I think aero coefficients are not correct in that case right as the AoA of the mesh is always zero? So it always requires one more step to get correct aero coefficients.

**Reply by ArneVoss** [reply]
Agreed, but I would say that Cx,y,z and Cm,x,y,z should be fine. If you seek for example Cl or Cd, you could calculate them afterwards by rotation of Cz and Cx about alpha.
